### PR TITLE
Java language migration updates

### PR DESCRIPTION
**Pull request by sman-81**
This pull request contains a number of Java language migration updates.
All changes are non-functional. They aim to improve code quality and make use of new Java language features.

- use Java functional primitives (instead of Guava's)
- use statement lambdas, use method references where possible
- avoid unnecessary boxing
- use Objects.equals
- use diamond operator without explicit type where permitted
- fix raw usage of parametrized classes

Tests are green.

### CODE REVIEW

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/transaction/internal/TransactionDecorator.java [hunk 1]
```diff
@@ -29,7 +30,7 @@
     public Handler decorateHandler(Handler base, Class<?> sqlObjectType, Method method) {
         final Transaction txnAnnotation = Stream.of(method, sqlObjectType)
                 .map(ae -> ae.getAnnotation(Transaction.class))
-                .filter(txn -> txn != null)
+                .filter(Objects::nonNull)
```
Comment (hgschmie) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/transaction/internal/TransactionDecorator.java:33]: this is one of those places where the explicit code is more readable than the function reference. Now I have to look up what "Objects::nonNull" does. The previous code was more readable.

File: core/src/main/java/org/jdbi/v3/core/locator/internal/ClasspathBuilder.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public ClasspathBuilder setExtension(@Nullable String extension) {
     }
 
     // org.foo.Bar$Inner -> org/foo/Bar$Inner
-    public ClasspathBuilder appendFullyQualifiedClassName(@Nonnull Class clazz) {
+    public ClasspathBuilder appendFullyQualifiedClassName(@Nonnull Class<?> clazz) {
```
Comment (hgschmie) [inline core/src/main/java/org/jdbi/v3/core/locator/internal/ClasspathBuilder.java:45]: I think this change is ok, but it does change a signature (from raw type to any type). @stevenschlansker to double check? It is an internal class, though, so it should be fine.

**Review by hgschmie — APPROVED** [review]

**Comment by hgschmie** [comment]
First, thank you for that PR! Most changes are cosmetic; I appreciate that this makes our code base more idiomatic Java 8+ code. 

I glanced at the test changes; those all seem good. I wrote a couple of comments on the actual code, again all seems fine. I will try to get the LGTM tests to finish (they crashed halfway through, which seems independent of the actual PR) and merge later today.

Did you hand-make those changes or did you use some tool? I know of people that use the modernizer plugin (https://github.com/gaul/modernizer-maven-plugin) but I had very mixed results with it.

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/transaction/internal/TransactionDecorator.java (hunk 1 above)
Comment (sman-81) [reply sqlobject/src/main/java/org/jdbi/v3/sqlobject/transaction/internal/TransactionDecorator.java:33]: it's a matter of taste :)
personally I like method references

File: core/src/main/java/org/jdbi/v3/core/locator/internal/ClasspathBuilder.java (hunk 2 above)
Comment (sman-81) [reply core/src/main/java/org/jdbi/v3/core/locator/internal/ClasspathBuilder.java:45]: Of course we can remove this change if you feel unsure.

**Comment by sman-81** [comment]
> First, thank you for that PR! Most changes are cosmetic; I appreciate that this makes our code base more idiomatic Java 8+ code.
> 
> I glanced at the test changes; those all seem good. I wrote a couple of comments on the actual code, again all seems fine. I will try to get the LGTM tests to finish (they crashed halfway through, which seems independent of the actual PR) and merge later today.
> 
> Did you hand-make those changes or did you use some tool? I know of people that use the modernizer plugin (https://github.com/gaul/modernizer-maven-plugin) but I had very mixed results with it.

Though I mostly use Eclipse, I have the habit of checking my projects using the `Code > Inspect Code` feature of IDEA.
This is a really well done and offers good pointers and insights, even though the refactorings most be applied with caution.

**Comment by sman-81** [comment]
Hi @hgschmie, @stevenschlansker
all above checks are green and the commit has no conflicts with master.
Would you like me to make any amendments?
or could you merge this PR?

**Comment by hgschmie** [comment]
sorry, got distracted. merged.

File: core/src/main/java/org/jdbi/v3/core/locator/internal/ClasspathBuilder.java (hunk 2 above)
Comment (stevenschlansker) [reply core/src/main/java/org/jdbi/v3/core/locator/internal/ClasspathBuilder.java:45]: Internal APIs are OK to change without regard for compatibility.

**Comment by stevenschlansker** [comment]
Changes look OK to me

**Comment by sman-81** [comment]
Thanks for merging.

**Pull request merged**